R/zchunk_LB134.Diet_Rfao.R

```diff
@@ -0,0 +1,384 @@
+    # Drop unnecessary composite regions from FAO diet table
+    # Original lines 71-85
+    FAO2050_Diet %>%
+      filter(FAO2050_reg %in% L134.Food_t_ctry_Dmnd_Y$FAO2050_reg) %>%
+      # Calculate FAO2050 diet for specified diet years
+      # Use linear interpolation to convert FAO2050 model time periods to GCAM "diet years"
+      group_by(FAO2050_reg, FAO2050_item) %>%
+      # Extend years in dataset to include DIET_YEARS
+      complete(year = unique(c(.$year, aglu.DIET_YEARS))) %>%
+      mutate(value = approx_fun(year, value)) %>%
+      # Add caloric content and demand category
+      left_join(select(FAO2050_items_cal, FAO2050_item, GCAM_demand, kcalkg, conv_d), by = "FAO2050_item") ->
+      L134.Diet_Rfao_Cfao_Yfao
+
+    # Build new table with only diet years subsetted
+    # Multiply through by caloric contents to get all measures in kcal/pers/d, by FAO region and food categories
+    # Original lines 87-92
+    L134.Diet_Rfao_Cfao_Yfao %>%
+      filter(year %in% aglu.DIET_YEARS) %>%
+      mutate(value = value * kcalkg / conv_d) %>%
+      select(FAO2050_reg, FAO2050_item, GCAM_demand, year, value) ->
+      L134.Diet_kcald_Rfao_Cfao_Y
+
+    # Aggregate by GCAM demand and FAO region, and compute future diet ratios by FAO2050 region
+    # Original lines 94-104
+    L134.Diet_kcald_Rfao_Cfao_Y %>%
+      filter(year %in% aglu.DIET_YEARS, !is.na(GCAM_demand)) %>%
+      group_by(FAO2050_reg, GCAM_demand, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup %>%
+      mutate(GCAM_demand = if_else(GCAM_demand == "total", "crops", GCAM_demand)) ->
+      L134.Diet_kcald_Rfao_Dmnd_Y
+
+    # Note on this next section from Page:
+    # Can't just use exogenous caloric contents for the crop and meat commodities, which we're estimating
+    # as averages without knowing the exact composition of the commodities (e.g., what portion of "cereals"
+    # is corn vs wheat in each region, as they have different caloric contents), and still hit the target
+    # reported "Total food (kcal/person/day)" in each region and time period. There were three options:
+    # (1) assign the crop caloric contents exogenously and allow the meat caloric contents to float;
+    # (2) vice versa; or
+    # (3) assign both crop and meat caloric contents, and then scale them all to hit the target value.
+    # It looks like I went with (2), maybe because I figured that the meat commodities probably have less
+    # heterogeneity in average caloric contents, and maybe also because it happened to be the easiest to implement.
+
+    # Compute diet ratios based on meat caloric contents and demand (option 2 above)
+    L134.Diet_kcald_Rfao_Dmnd_Y %>%
+      filter(GCAM_demand == "meat") %>%
+      select(-GCAM_demand) %>%
+      rename(meat_value = value) %>%
+      right_join(L134.Diet_kcald_Rfao_Dmnd_Y, by = c("FAO2050_reg", "year")) %>%
+      mutate(value = if_else(GCAM_demand == "crops", value - meat_value, value)) %>%
```

Comment:
This is why I'm not a big fan of using `value` as a column name for intermediate tables.  I usually give columns meaningful names, and then convert to `value` at the same time I add metadata for the final output.